### ISSUE TITLE
No template found issue for C-terminal residues

### ISSUE THREAD

**Issue by okbckim** [openmm/openmm#4530]
Hi there,

I am having issue to make system from the protein I attached. The error occurs in forcefield.createSystem with error message:

ValueError: No template found for residue 210 (LEU).  The set of atoms matches CLEU, but the bonds are different.  For more information, see https://github.com/openmm/openmm/wiki/Frequently-Asked-Questions#template

I checked the corresponding residue, 208 LEU, in the file. Everything looks good visually in pymol. The C-terminal of original structure was capped with a hydrogen. I use Pymol Builder to change the hydrogen to O- and its atom name to OXT. 

Code to reproduce is

from openmm.app import *
from openmm import *
from openmm.unit import *
from sys import stdout

pdb = PDBFile('a2a_hip278_protein_terminal.pdb')

forcefield = ForceField('amber14-all.xml', 'amber14/tip3pfb.xml')

system = forcefield.createSystem(pdb.topology, nonbondedMethod=PME,
        nonbondedCutoff=1*nanometer, constraints=HBonds)
        
It will be greatly appreciated if you can help. I spent so much time to figure out but couldn't solve it.

Note I add .txt to the pdb file to load to this case.
[a2a_hip278_protein_terminal.pdb.txt](https://github.com/openmm/openmm/files/15178870/a2a_hip278_protein_terminal.pdb.txt)

**Comment by peastman** [comment]
Your PDB file has some problems.  Here's the relevant section.

```
ATOM   3180  N   LEU A 208      16.948   7.178 -25.739  1.00  0.00           N  
ATOM   3181  CA  LEU A 208      16.310   8.466 -25.974  1.00  0.00           C  
ATOM   3182  C   LEU A 208      16.648   9.279 -24.730  1.00  0.00           C  
ATOM   3183  O   LEU A 208      16.170  10.374 -24.556  1.00  0.00           O  
ATOM   3184  CB  LEU A 208      14.753   8.364 -26.231  1.00  0.00           C  
ATOM   3185  CG  LEU A 208      13.752   7.984 -25.121  1.00  0.00           C  
ATOM   3186  CD1 LEU A 208      12.210   8.109 -25.515  1.00  0.00           C  
ATOM   3187  CD2 LEU A 208      14.057   6.586 -24.499  1.00  0.00           C  
ATOM   3188  OXT LEU A 208      17.557   8.801 -23.735  1.00  0.00           O1-
ATOM   3189  H   LEU A 208      17.135   7.052 -24.755  1.00  0.00           H  
ATOM   3190  HA  LEU A 208      16.620   8.874 -26.937  1.00  0.00           H  
ATOM   3191  HB2 LEU A 208      14.416   9.385 -26.406  1.00  0.00           H  
ATOM   3192  HB3 LEU A 208      14.541   7.805 -27.142  1.00  0.00           H  
ATOM   3193  HG  LEU A 208      13.901   8.670 -24.287  1.00  0.00           H  
ATOM   3194 HD11 LEU A 208      11.802   9.060 -25.173  1.00  0.00           H  
ATOM   3195 HD12 LEU A 208      11.996   8.144 -26.583  1.00  0.00           H  
ATOM   3196 HD13 LEU A 208      11.586   7.305 -25.125  1.00  0.00           H  
ATOM   3197 HD21 LEU A 208      14.381   5.925 -25.303  1.00  0.00           H  
ATOM   3198 HD22 LEU A 208      14.820   6.631 -23.722  1.00  0.00           H  
ATOM   3199 HD23 LEU A 208      13.164   6.208 -24.001  1.00  0.00           H  
ATOM   3200  N   GLU A 219       9.932   6.197 -38.597  1.00  0.00           N  
ATOM   3201  CA  GLU A 219      10.440   6.597 -37.279  1.00  0.00           C  
ATOM   3202  C   GLU A 219       9.540   6.216 -36.004  1.00  0.00           C  
```

Some things to notice here.

- The residue ID jumps from 208 to 219, suggesting there are probably some missing residues.
- There's a big gap in space.  Atom C of LEU 208 and atom N for GLU 219 are close to 20 A apart.  That also suggests there are missing residues, and those atoms should be bonded together as a single chain.
- The LEU has been terminated as if it were the end of a chain.  It contains an OXT atom, which is only found in C-terminal residues.
- But contrary to all of the above, it's all one chain.  There's no `TER` record to indicate a new chain, and the chain ID remains unchanged as `A`.

**Comment by okbckim** [comment]
Thank you so much for your comprehensive comments.

I changed Chain ID from A to B or C when missing loops exist. With this simple change as you suggested, I am able to run the simulation. Great suggestion!

**Comment by peastman** [comment]
Glad that worked!

**Comment by okbckim** [comment]
Can I ask you one more question related to the subject.
I call modeller.addMembrane to add POPC membrane. I was able to run the simulation with the system.
Then, I built membrane using another software. The POPC molecule looked same except coordinates. However, I encountered the error message. 284 POP is the first POPC molecule in the system.

ValueError: No template found for residue 284 (POP).  The set of atoms matches POPC, but the bonds are different.  Perhaps the chain is missing a terminal group? I attached the pdb file from this case.

Thank you again!
[system.pdb.txt](https://github.com/openmm/openmm/files/15181314/system.pdb.txt)
